scheduler/task.go

```diff
@@ -537,6 +527,36 @@ func (t *task) fire() {
 	t.state = core.TaskSpinning
 }
 
+// disable proceeds disabling a task which consists of changing task state to disabled, emitting an appropriate event
+// and unsubscribing its dependencies
+func (t *task) disable(failureMsg string) {
+	t.Lock()
+	t.state = core.TaskDisabled
+	t.Unlock()
+
+	// Send task disabled event
+	event := new(scheduler_event.TaskDisabledEvent)
+	event.TaskID = t.id
+	event.Why = fmt.Sprintf("Task disabled with error: %s", failureMsg)
+	defer t.eventEmitter.Emit(event)
+
+	// We need to unsubscribe from deps when a task has disabled
+	t.UnsubscribePlugins()
+}
```

Comment:
FYI:
The only my inaccuracy is about handling errors from `UnsubscibePlugins()`.  I would like to add here log.Error at least. What do You think about that?

Notice that even in historical commit https://github.com/intelsdi-x/snap/blob/c592966ef49f7c1dbb78b712a6d3f802a7e00b8e/scheduler/scheduler.go#L806 there was no react if unsubscribing failed.